lib/datasource/aws-machine-image/readme.md

```diff
@@ -0,0 +1,117 @@
+> :warning: **This datasource is _experimental_**: Be aware that its syntax and behavior may change at any time!
+
+This datasource returns the latest [Amazon Machine Image](https://docs.aws.amazon.com/en_en/AWSEC2/latest/UserGuide/AMIs.html) via the AWS API (valid credentials required).
+
+Because there is no general lookupName, you have to use the [describe images filter](https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/clients/client-ec2/interfaces/describeimagescommandinput.html#filters) as minified JSON as a lookupName.
+
+Example:
+
+```yaml
+# Getting the latest official EKS image from AWS (account '602401143452' for eu-central-1) for EKS 1.21 (name matches 'amazon-eks-node-1.21-*') would look as a describe images filter like:
+
+[
+    {
+        "Name": "owner-id",
+        "Values": [
+            "602401143452"
+        ]
+    },
+    {
+        "Name": "name",
+        "Values": [
+            "amazon-eks-node-1.21-*"
+        ]
+    }
+]
```

Comment:
The YAML takes up a bit of horizontal space, can this be condensed in some way maybe?

Reply:
I changed it to a more compact style ...